src/NuGet.Core/NuGet.Configuration/Settings/SettingSection.cs

```diff
@@ -28,7 +44,7 @@ protected SettingSection(string name, IReadOnlyDictionary<string, string> attrib
                 throw new ArgumentException(Resources.Argument_Cannot_Be_Null_Or_Empty, nameof(name));
             }
 
-            ElementName = XmlConvert.EncodeLocalName(name);
+            ElementName = name;
```

Comment:
Was it regressed for sectionName? I see you were still encoding it...

Reply:
We did not see any scenario because we don't currently have any section with spaces... We are only encoding on creation, but not when comparing the name of the section to other strings, therefore the same bug could be a problem here if there was a section with a special character.